src/js/components/Button/StyledButton.js

```diff
@@ -206,11 +207,15 @@ const StyledButton = styled.button`
     props.theme.button &&
     props.theme.button.disabled &&
     disabledButtonStyle(props)}
-  
+
   &:focus {
     ${props => (!props.plain || props.focusIndicator) && focusStyle()}
   }
 
+  &:focus:not(:focus-visible) {
+    ${unfocusStyle()}
```

Comment:
When I click on a (such as the one with label "Default")  Button in [this story](https://deploy-preview-5148--sad-tereshkova-173d07.netlify.app/?path=/story/controls-button-basic--basic), I don't get the hover behavior unless I click outside the button (somewhere on the page). I would expect that I would still get hover behavior again after clicking.

Also, would we want to apply similar styling to StyledButtonKind so the same focus behavior applies to buttons with themes relying on `button.default`?

Reply:
cool, let me look where we apply the focus styling elsewhere. In theory, everywhere we use `:focus` we could augment it with the logic proposed in this PR.

I'm not sure I follow your first comment though. I would not expect this PR to affect the hover behavior at all. I'm only changing the focus behavior when using the mouse. Do you mind sharing more details?

Reply:
FYI I've updated this PR with the other components that needed the `:focus-visible` extension. I've skip input components like TextInput and FileInput.

Reply:
Summarizing from offline discussion: It seems that grommet does not apply `hover` styling if an element is in focus (existing behavior). I am wondering if a separate enhancement would be to always apply hover styling when moused over (whether in focus or not). This would resolve the behavior I noted in my initial comment.